js/debugger.ts

```diff
@@ -0,0 +1,59 @@
+import { debug, toHex } from './util';
+import { word } from './types';
+
+import SYMBOLS from './symbols';
+import CPU6502, { DebugInfo } from './cpu6502';
+
+const alwaysBreak = (_info: DebugInfo) => { return true; };
+
+export default class Debugger {
+    private TRACE = false;
+    private MAX_TRACE = 256;
+    private trace: DebugInfo[] = [];
+    private breakpoints: Map<number, (info: DebugInfo) => boolean> = new Map()
```

Comment:
I think this is a Map from address -> breakpoint function. Perhaps signal that by using `word` instead of `number`?